src/store/actions/wallet.ts

```diff
@@ -296,6 +301,78 @@ export const injectFakeTransaction = ({
 	}
 };
 
+/**
+ * Adds a new transfer transaction to the store.
+ * @param {TTransfer} payload
+ */
+export const addTransfer = (payload: TTransfer): void => {
+	dispatch({ type: actions.ADD_TRANSFER, payload });
+};
+
+export const updateTransfer = ({
+	txId,
+	type,
+	confirmations,
+}: {
+	txId: string;
+	type: ETransferType;
+	confirmations?: number;
+}): void => {
+	switch (type) {
+		case ETransferType.open: {
+			const orders = getBlocktankStore().orders;
+			const order = orders.find((o) => o.channel?.fundingTx.id === txId);
+			if (order) {
+				const paymentTxId = order.payment.onchain.transactions[0].txId;
+				dispatch({
+					type: actions.UPDATE_TRANSFER,
+					payload: { txId: paymentTxId },
+				});
+			}
+			break;
+		}
+		case ETransferType.coopClose:
+		case ETransferType.forceClose: {
+			dispatch({
+				type: actions.UPDATE_TRANSFER,
+				payload: { txId, confirmations },
+			});
+			break;
+		}
+	}
+};
+
+export const updatePendingTransfers = (headerHeight: number): void => {
+	const { currentWallet, selectedNetwork } = getCurrentWallet();
+	const transactions = currentWallet.transactions[selectedNetwork];
+	const transfers = currentWallet.transfers[selectedNetwork];
+	const pendingTransfers = transfers.filter((t) => {
+		return (
+			t.type !== ETransferType.open && t.status === ETransferStatus.pending
+		);
+	}) as TTransferToSavings[];
+
+	pendingTransfers.forEach((transfer) => {
+		const tx = transactions[transfer.txId];
+		if (transfer.confirmations <= 6) {
+			const confs = tx.height === -1 ? 1 : headerHeight - tx.height + 1;
```

Comment:
```suggestion
			const confs = tx.height < 1 ? 0 : headerHeight - tx.height + 1;
```
Seeing a confirmation miscalculation. Possibly from this snippet.
<img width="662" alt="Screenshot 2024-02-10 at 12 48 03 PM" src="https://github.com/synonymdev/bitkit/assets/8532651/587d93cc-2edb-4745-9069-5d1796610852">